packages/trader/src/Modules/Account/Sections/Security/ConnectedApps/connected-apps.jsx

```diff
@@ -0,0 +1,132 @@
+import getConnectedAppsColumnsTemplate from './data-table-template';
+import Loading from '../../../../../templates/_common/components/loading.jsx';
+
+class ConnectedApps extends React.Component {
+    state = {
+        is_loading: true,
+        is_modal_open: false,
+        selected_app_id: null,
+    };
+
+    componentDidMount() {
+        this.updateApp();
+    }
+
+    handleRevokeAccess = () => {
+        this.setState({ is_modal_open: false, is_loading: true });
+        this.props.revokeAccess(this.state.selected_app_id).then(() => this.updateApp());
+    };
+
+    handleToggleModal = (app_id = null) => {
+        if (this.state.is_modal_open === true) {
+            this.setState({ is_modal_open: false, selected_app_id: null });
+        } else {
+            this.setState({ is_modal_open: true, selected_app_id: app_id });
+        }
+    };
+
+    updateApp = () => {
+        this.props
+            .fetchConnectedApps()
+            .then(result => {
+                this.props.setConnectedApps(result);
+                this.setState({ is_loading: false });
+            })
+            .catch(() => {
+                this.setState({ is_loading: true });
+            });
+    };
+
+    columns_map = getConnectedAppsColumnsTemplate(this.handleToggleModal).reduce((map, item) => {
+        map[item.col_index] = item;
+        return map;
+    }, {});
+
+    mobileRowRenderer = ({ row }) => {
+        return (
+            <div className='data-list__row'>
+                <div className='data-list__col'>
+                    <DataList.Cell row={row} column={this.columns_map.name} />
+                    <DataList.Cell row={row} column={this.columns_map.scopes} />
+                </div>
+                <div className='data-list__col--small'>
+                    <DataList.Cell row={row} column={this.columns_map.last_used} />
+                    <DataList.Cell row={row} column={this.columns_map.app_id} is_footer={true} />
+                </div>
+            </div>
+        );
+    };
+
+    render() {
+        return (
+            <section className='connected-apps'>
+                <p className='connected-apps__title'>{localize('Authorised applications')}</p>
+                <div>
+                    {this.state.is_loading ? (
+                        <Loading />
+                    ) : (
+                        <>
+                            <DesktopWrapper>
+                                <DataTable
+                                    className='connected-apps'
+                                    data_source={this.props.connected_apps}
+                                    columns={getConnectedAppsColumnsTemplate(this.handleToggleModal)}
+                                    custom_width={'100%'}
+                                    getRowSize={() => 56}
+                                    is_empty={false}
+                                >
+                                    {this.state.is_loading && <Loading />}
```

Comment:
as I understand this part of the code won't be reached if state is loading, right?

Reply:
No, but I pass a loading component to it just like the other data table components

Reply:
Yeah we can remove lines 71:73 and rely on Datatable's own loading state. Since right now, if `this.state.is_loading` the fragment won't load at all and line 84 is redundant.